### PR TITLE
DU-550: changes for funnel reports

### PR DESCRIPTION
**Pull request by ddevadat**
Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

### Type of change

Please choose appropriate options.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

### How Has This Been Tested?

Please describe the tests that you ran to verify your changes in the below checkboxes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Ran Test A
- [ ] Ran Test B

**Test Configuration**:
* Software versions:
* Hardware versions:

### Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: data-products/src/main/scala/org/sunbird/analytics/sourcing/FunnelReport.scala [hunk 1]
```diff
@@ -43,6 +43,7 @@ object FunnelReport extends IJob with BaseReportsJob {
   val jobName = "Funnel Report Job"
   val db = AppConf.getConfig("postgres.db")
   val url = AppConf.getConfig("postgres.url") + s"$db"
+  val globalContainer = AppConf.getConfig("cloud.container.reports")
```
Comment (ddevadat) [inline data-products/src/main/scala/org/sunbird/analytics/sourcing/FunnelReport.scala]: looks like a redundant variable

File: data-products/src/main/scala/org/sunbird/analytics/sourcing/FunnelReport.scala [hunk 2]
```diff
@@ -122,7 +123,15 @@ object FunnelReport extends IJob with BaseReportsJob {
       .drop("channel","id","program_id")
       .persist(StorageLevel.MEMORY_ONLY)
 
-    val storageConfig = getStorageConfig("reports", "")
+    // support to other CSP and report container should be read from configuration
+    // val storageConfig = getStorageConfig("reports", "")
+    val modelParams = config.modelParams.getOrElse(Map[String, Option[AnyRef]]());
+    val container = modelParams.getOrElse("storageContainer", "reports").asInstanceOf[String]
+    print("container for Funnel Report: " + container)
```
Comment (ddevadat) [inline data-products/src/main/scala/org/sunbird/analytics/sourcing/FunnelReport.scala]: remove print and loggers

File: data-products/src/main/scala/org/sunbird/analytics/sourcing/FunnelReport.scala [hunk 3]
```diff
@@ -122,7 +123,15 @@ object FunnelReport extends IJob with BaseReportsJob {
...
+    JobLogger.log(s"saving report in bucket - ${container}", None, Level.INFO)
+    // use global container for the report - this is only a fallback if required
```
Comment (ddevadat) [inline data-products/src/main/scala/org/sunbird/analytics/sourcing/FunnelReport.scala]: disabled code ( commented)  can be removed before raising the upstream pr

File: data-products/src/main/scala/org/sunbird/analytics/sourcing/FunnelReport.scala (hunk 1 above)
Comment (heungheung) [reply data-products/src/main/scala/org/sunbird/analytics/sourcing/FunnelReport.scala]: the value is not required

File: data-products/src/main/scala/org/sunbird/analytics/sourcing/FunnelReport.scala (hunk 2 above)
Comment (heungheung) [reply data-products/src/main/scala/org/sunbird/analytics/sourcing/FunnelReport.scala]: ok

**Review by heungheung — APPROVED** [review]
clean up code and ready for merge

**Pull request merged**